File: mod/library/server/handle-message/exporter/process_stake.lua
local check_for_override = SMODS.load_file("mod/server/handle-message/exporter/_common/check_for_override.lua")()
local output_image = SMODS.load_file("mod/server/handle-message/exporter/_common/output_image.lua")()

local function process_stake(sets, stake)
	local item = {}
	output_image(stake)
	item.key = stake.key
	item.name = localize { type = 'name_text', set = 'Stake', key = stake.key }
	local loc_vars = nil
	if stake.loc_vars and type(stake.loc_vars) == 'function' then
		local res = stake:loc_vars() or {}
		loc_vars = res.vars or {}
	end
	if stake.mod and stake.mod.id ~= "Aura" and stake.mod.id ~= "aure_spectral" then
		item.mod = stake.mod.id
	end
	item.description = localize { type = 'raw_descriptions', key = stake.key, set = "Stake", nodes = {}, vars = loc_vars }
	item.image_url = check_for_override("images/" .. stake.key:gsub("?", "_") .. ".png")
	if item.name then
		sets["Stake"][item.key] = item
	end
end

return process_stake

File: mod/library/server/handle-message/exporter/process-stake.ts
import {
	outputImage
} from "./_common/_exports.ts";

/**
 * Process stake data and add to sets
 *
 * @param stake - The stake object to process
 */
const processStake = (stake: typeof SMODS.Stake) => {
	const item: Record<string, any> = {};

	outputImage({
		...stake,
		atlas: stake.atlas ?? "chips",
		set: "Stake"
	});

	item.key = stake.key;
	item.name = localize({
		key: stake.key,
		set: "Stake",
		type: "name_text"
	});

	let localizationVariables = null;

	if (stake.loc_vars && typeof stake.loc_vars === "function") {
		const result = stake.loc_vars() || {};

		localizationVariables = result.vars || {};
	}

	if (stake.mod && stake.mod.id !== "Aura" && stake.mod.id !== "aure_spectral") {
		item.mod = stake.mod.id;
	}

	item.description = localize({
		key: stake.key,
		nodes: [],
		set: "Stake",
		type: "raw_descriptions",
		vars: localizationVariables
	});

	item.image_url = `${stake.key.replaceAll("?", "_")}.png`;

	if (item.name) {
		return item;
	}
};

export default processStake;
